django_cloud_tasks/middleware/pubsub_headers_middleware.py

```diff
@@ -51,7 +52,7 @@ def extract_headers(self, request) -> dict[str, Any]:
             return {}
 
         headers = {}
-        for key, value in message.attributes.items():
-            if key.startswith(self.pubsub_header_prefix):
-                headers[key.removeprefix(self.pubsub_header_prefix)] = value
+        message_headers = message.data.get(self.propagated_headers_key) or {}
+        for key, value in message_headers.items():
+            headers[key.removeprefix(self.pubsub_header_prefix)] = value
```

Comment:
I think the `self.pubsub_header_prefix` adds no value now that you have a dedicated key to store all the headers.